050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,801 @@
+It is therefore left to the operator to detect if there are multiple `KafkaTopics` which point to the same topic in Kafka.
+
+To do this, the operator will keep an in-memory mapping of _topic name_ to (`metadata.namespace`, `metadata.name`)-pairs (where _topic name_ is `spec.topicName`, defaulting to `metadata.name` as described above).
+This will allow us to detect the case where multiple resources are attempting to manage the same topic in Kafka.
+When this happens the unique oldest (as determined by the `metadata.creationTimestamp`) `KafkaTopic` will be considered to manage the topic, and the other `KafkaTopics` will be updated: their `Ready` status will change to `False` with suitable `reason`.
+
+```yaml=
+kind: KafkaTopic
+metadata:
+  generation: 123
+  namespace: some-namespace
+  name: foo
+  creationTimestamp: 20230301T103000Z
+status:
+  topicName: __consumer_offsets
+  observedGeneration: 123
+  conditions:
+  - type: Ready
+    status: True
+    lastTransitionTime: 20230301T103000Z
+---
+kind: KafkaTopic
+metadata:
+  generation: 456
+  namespace: some-namespace
+  name: bar
+  creationTimestamp: 20230420T103000Z
+status:
+  topicName: __consumer_offsets
+  observedGeneration: 456
+  conditions:
+  - type: Ready
+    status: False
+    reason: ResourceConflict
+    message: Managed by some-namespace/foo
+    lastTransitionTime: 20230420T103000Z
+```
+
+### Unidirectionality
+
+As mentioned, we would only synchronize topic state from Kube to Kafka.
+Users and applications could create, delete and modify topics using the Kafka protocol (e.g. via the `Admin` client) which would **not** be reflected by a `KafkaTopic` in Kube.
+
+In other words, the existence of at least some `KafkaTopic` resources would be decoupled from the existence of a corresponding topic in Kafka.
+While users could choose to create a topic in Kafka by creating a `KafkaTopic`, they could also create, modify and eventually delete topics directly without a `KafkaTopic` ever existing.
+
+When a `KafkaTopic` did exist for a given topic in Kafka then any changes made via other tooling would eventually be reverted.
+
+### "Managed" and "unmanaged" topics
+
+We will refer to a topic which exists in a Kafka cluster and has a corresponding `KafkaTopic` as a _managed topic_.
+A topic which exists in a Kafka cluster without a matching `KafkaTopic` is an _unmanaged topic_.
+
+Naturally any topic created due to the creation of a `KafkaTopic` is managed from the outset.
+
+---
+
+#### Example 1: `KafkaTopic` creation when topic does NOT exist in Kafka
+
+> 1. The user creates a `KafkaTopic`:
+> 
+>     ```yaml=
+>     metadata:
+>       generation: 1
+>       name: foo
+>     spec:
+>       partitions: 12
+>       replicas: 50
+>    ```
+> 
+> 2. The operator checks its in-memory map to see whether `foo` is already managed by some other resource.
+> 
+>     1. If so, then both resources are updated with error conditions in their `status`.
```

Comment:
I forgot to update this bit when I changed this from "both get put into `ResourceConflict`" (which is what I originally had) to "only the newest resource gets put into `ResourceConflict`" aka "first in wins" (which was suggested by @fvaleri). There are pros and cons either way:

Marking both as conflicted:
* means that the topic is temporarily unmanageable via either resource
* means that 
    * either the operator needs to update another resource than the one that changed, complicating the locking which we use to prevent concurrent reconciliation in the case of multi-threaded reconciliation.
    * or that we don't try to update the status of "the other" resource when reconciling "this one", leaving it to the next timed reconciliation to update the other one. The lagginess of this simpler alternative is not attractive.
* means that the error message on the condition needs to enumerate every conflicting resource, which could actually be many more than 2 in the general case.

Marking only the later-created resources is actually simpler, since updates to other resources are avoided and the conflicted resources need only point to the first-created resource. These are all relatively shallow considerations, but together they persuaded me to make the change.